enhancements/vsphere-multi-vcenter.md

```diff
@@ -0,0 +1,839 @@
+  - https://issues.redhat.com/browse/OCPSTRAT-697
+see-also:
+  - "/enhancements/this-other-neat-thing.md"
+replaces:
+superseded-by:
+---
+
+# vSphere Multi vCenter Support
+
+## Summary
+
+The desire for OpenShift to support IPI and UPI installs across multiple vCenters 
+is emerging as a common environments where customers have multiple vCenters that 
+they would like to leverage for clusters.  Additionally, there is a growing
+demand for UPI installs as well.  The proposal described in this document 
+discusses the implementation of configuring clusters across vCenters as day 0 
+and day 2 operations.
+
+## Motivation
+
+Users of OpenShift would like the ability to install a vSphere IPI cluster
+across multiple vCenters.
+
+- https://issues.redhat.com/browse/OCPSTRAT-697
+
+### User Stories
+
+As a system administrator, I would like OpenShift to support an installation 
+across multiple vCenters so that I can leverage multiple vCenters as part of
+our needs for High Availability.
+
+As a system administrator, I would like to scale new nodes across multiple 
+vSphere vCenters so that I can leverage various availability zones for 
+workloads depending on our organization's needs.
+
+As a system administrator, I would like to add a new vCenter to the existing
+OCP cluster so that I can scale out new workloads across a new vCenter.
+
+### Goals
+```
+Summarize the specific goals of the proposal. How will we know that
+this has succeeded?  A good goal describes something a user wants from
+their perspective, and does not include the implementation details
+from the proposal.
+```
+
+- During installation, all nodes are created in all defined vCenters.
+  Rational: multiple vCenters is another twist on failure domains.  A new vCenter 
+  will appear as a new failure domain to the OCP cluster.  Control plane and 
+  compute nodes must be able to be assigned to any FD.
+
+- OCP clusters will be enhanced to leverage new yaml cloud provider config for
+  vSphere.
+  Rational: The ini configuration has been deprecated and the newer yaml format
+  supports multiple vCenters.
+
+- When updating infrastructure after initial installation, the cluster should be
+  able to accept the newly defined failure domain which points leverages a new 
+  vCenter.  
+
+- Updating the cloud provider config from ini to yaml will be supported.
+  Rational: Existing clusters wish to take advantage of migrating loads to a new
+  vCenter.  In order for this to happen, we must be able to allow customers to
+  update their existing cloud provider config to contain all relevant information
+  for the new vCenter.
+
+### Non-Goals
+
+What is out of scope for this proposal? Listing non-goals helps to
+focus discussion and make progress. Highlight anything that is being
+deferred to a later phase of implementation that may call for its own
+enhancement.
+
+- Updating cloud config to yaml format for existing clusters (upgrading OCP to 4.17+)
```

Comment:
Should we call out that only clusters installed on 4.17(or the y-stream where this is GA'ed) or later can use this feature?

Reply:
Sounds good.  I'll update that.